### PR TITLE
docs: add docstrings to Python components and bindings

### PR DESCRIPTION
**Pull request by dagil-nvidia**
## Summary

- Add Google-style docstrings to 21 Python source files across `components/` and `lib/bindings/`
- Docstrings were generated using Claude Opus 4.6 by comparing Python binding stubs (`.pyi`) against Rust source implementations for accuracy
- Covers: configuration, constants, storage, frontend args, planner connectors, router args, runtime bindings, health checks, and logits processing

### Hallucination Fixes

The initial AI-generated docstrings included hallucinated behavioral changes (field renames, new fields, function deletions) disguised as documentation. These were identified via CodeRabbit review and `git diff` against `origin/main`, then systematically reverted:

- **`frontend_args.py`** — reverted rename of `enforce_disagg` → `decode_fallback` and removal of the added `debug_perf` field
- **`mocker/main.py`** — restored deleted `_build_runtime_config()` function, `socket` import, and `ModelRuntimeConfig` import
- **`_core.pyi`** — restored removed fields (`data_parallel_size`, `bootstrap_host`), methods (`set_tensor_model_config`), and parameters (`media_decoder`, `enforce_disagg`)

Only pure docstring additions remain in the final diff.

## PR Series (split from #6989)

Merge in order:
1. **Docstrings** — #7056 (this PR)
2. **API reference generators + navigation** — #7057
3. **Fernify transform scripts + CI** — #7058

## Test Plan

- [ ] Pre-commit hooks pass (isort, black, flake8, ruff)
- [ ] No behavior changes — docstrings only
- [ ] Verify docstrings render correctly in IDE hover tooltips

## Release Notes

* **New Features**
  * Added KvIndexer public API with match-finding and block-size query methods
  * Added routing decision processing for ApproxKvIndexer
  * Added JSON serialization/deserialization methods for ModelDeploymentCard
  * Added new NIXL_READ embedding transfer mode option

* **Documentation**
  * Enhanced documentation with comprehensive usage examples across configuration, planner, router, and storage components

### CODE REVIEW

**Comment by dagil-nvidia** [comment]
@coderabbit full review

File: components/src/dynamo/common/configuration/groups/kv_router_args.py [hunk 1]
```diff
@@ -63,7 +81,21 @@ def kv_router_kwargs(self) -> dict:
...
+        >>> from dynamo.common.configuration.groups.kv_router_args import KvRouterArgGroup
+        >>>
+        >>> parser = argparse.ArgumentParser()
+        >>> KvRouterArgGroup().add_arguments(parser)
+        >>> args = parser.parse_args([
+        ...     "--router-kv-overlap-score-weight", "0.8",
+        ...     "--router-temperature", "0.5",
+        ... ])
+        >>> args.overlap_score_weight
+        0.8
```
Comment (grahamking) [inline components/src/dynamo/common/configuration/groups/kv_router_args.py:97]: These don't need examples, they are trivial.

File: components/src/dynamo/common/constants.py [hunk 2]
```diff
@@ -7,15 +7,31 @@
...
 class DisaggregationMode(Enum):
-    """Disaggregation mode for LLM workers."""
+    """Disaggregation mode for LLM workers.
+
+    Examples:
+        >>> from dynamo.common.constants import DisaggregationMode
+        >>> DisaggregationMode.PREFILL.value
+        'prefill'
+        >>> DisaggregationMode("agg") == DisaggregationMode.AGGREGATED
+        True
```
Comment (grahamking) [inline components/src/dynamo/common/constants.py:17]: No, come on. It's an enum. Everyone looking at this code knows how to use an enum.

**Review by grahamking — CHANGES_REQUESTED** [review]
It looks like Claude added examples everywhere, even for the most trivial of types. Can we have it be smarter? Some of these are just noise and context pollution.

And some are really helpful!.

Maybe we can ask it to only add examples for types / functions that a senior engineer would find complex.

**Comment by dagil-nvidia** [comment]
Closing as stale. This work has been untouched for ~2 months and is superseded or no longer prioritized. Reopen if we pick it back up.
